### ISSUE TITLE
Static nodeport configuration  for ingress resourse 

### ISSUE THREAD

**Issue by izutheo** [appscode/voyager#400]
Hi all,
 I  just pull the latest version with the static nodeport assignment functionality  fixed.  Unfortunately i still cannot configure the ingress resource to assign a specific port to the type: nodeport. Is there a documentation for this ?

**Comment by tamalsaha** [comment]
Hi, this feature has been implemented in master and will be part of 3.2.0 release. We are yet to update the docs. If you want to test it out, you can use the image tag `appscode/voyager:3.2.0-beta.2` . This is what an example Ingress looks like:
```

**Comment by izutheo** [comment]
Perfect thank you very much

**Comment by izutheo** [comment]
Hi tamalsaha,

I think i might jumped the gone on this, but i was actually asking if there was a way to set a static port for the ingress resource above and not have the port change every time i update the ingress resource.


so following your example above, If i updated the ingress resource with an additional path 



apiVersion: voyager.appscode.com/v1beta1
kind: Ingress
metadata:
  name: test
  namespace: default
  annotations:
    ingress.appscode.com/type: NodePort
spec:
  rules:
  - host:
    http:
      paths: /test
      - backend:
          serviceName: test-server
          servicePort: 80
          nodePort: 32068
    paths: /testing
      - backend:
          serviceName: test-server-2
          servicePort: 80
  So i am basically asking for the ability to have one static port for entry into the cluster and not to a service.

**Comment by tamalsaha** [comment]
@izutheo , sorry I misspoke last time. You can only set one static port for each HAProxy port (which is the entry to the cluster). If you look at the Go struct, the `nodePort` is set for a host:port combination.
```
apiVersion: voyager.appscode.com/v1beta1
kind: Ingress
metadata:
  name: test
  namespace: default
  annotations:
    ingress.appscode.com/type: NodePort
spec:
  rules:
  - host:
    http:
      nodePort: 32068
      paths:
      - path: /t1
        backend:
          serviceName: t1-server
          servicePort: 80
      - path: /t2
        backend:
          serviceName: t2-server
          servicePort: 80
```

**Comment by tamalsaha** [comment]
I have pushed a new image with tag `appscode/voyager:3.2.0-beta.3` , this has some important fixes from (#394). Please try with that.

**Comment by izutheo** [comment]
Thank you very much

**Comment by colltrix35** [comment]
@tamalsaha 

I am having the same issue as the one documented above. Instead of having one port per host via nodeport port  and having multiple static ports as the number of host increases. 

is it possible to have one static port for all host.
using your example on github,



apiVersion: voyager.appscode.com/v1beta1
kind: Ingress
metadata:
  name: haproxy
  namespace: default
spec:
  backend:
    headerRule:
    - X-Forwarded-Host %[req.hdr(Host)]
    - X-Forwarded-Port %[dst_port]
    - X-Forwarded-Proto https
    serviceName: phabricator
    servicePort: "80"
  rules:
  - host: maven.example.com
    http:
      paths:
      - backend:
          rewriteRule:
          - ^([^\ :]*)\ /(.*)$ \1\ /maven/\2
          serviceName: artifact-server
          servicePort: "80"
  - host: composer.example.com
    http:
      paths:
      - backend:
          rewriteRule:
          - ^([^\ :]*)\ /(.*)$ \1\ /composer/\2
          serviceName: artifact-server
          servicePort: "80"
  - host: npm.example.com
    http:
      paths:
      - backend:
          rewriteRule:
          - ^([^\ :]*)\ /(.*)$ \1\ /npm/\2
          serviceName: artifact-server
          servicePort: "80"
  - host: docker.example.com
    http:
      paths:
      - backend:
          rewriteRule:
          - ^([^\ :]*)\ /(.*)$ \1\ /docker/\2
          serviceName: artifact-server
          servicePort: "80"
  - host: api.example.com
    http:
      paths:
      - backend:
          serviceName: authserver
          servicePort: "9021"
        path: /kubernetes/v1beta1/webhooks
  - host: auth.example.com
    http:
      paths:
      - backend:
          serviceName: example-loginserver
          servicePort: "7070"
        path: /oauth
  - host: auth.example.com
    http:
      paths:
      - backend:
          headerRule:
          - X-Client-IP %[src]
          rewriteRule:
          - ^([^\ ]*)\ /client-ip/(.*)     \1\ /\2
          serviceName: client-ip
          servicePort: "60010"
        path: /client-ip
  - host: '*.example.info'
    http:
      paths:
      - backend:
          serviceName: aphlictserver
          servicePort: "22280"
        path: /_org/ws
  - host: api.example.com
    http:
      paths:
      - backend:
          rewriteRule:
          - ^([^\ :]*)\ /(.*)$ \1\ /_org/api/\2
          serviceName: example-apis
          servicePort: "9866"
        path: /onebox
  - host: api.example.com
    http:
      paths:
      - backend:
          rewriteRule:
          - ^([^\ :]*)\ /(.*)$ \1\ /_org/api/\2
          serviceName: apiserver
          servicePort: "9877"
  - host: '*.example.info'
    http:
      paths:
      - backend:
          serviceName: apiserver
          servicePort: "9877"
        path: /_org/api
  - host: '*.example.info'
    http:
      paths:
      - backend:
          serviceName: ui-server
          servicePort: "80"
        path: /artifactory
  - host: '*.example.info'
    http:
      paths:
      - backend:
          serviceName: ui-server
          servicePort: "80"
        path: /ui-server
  - host: '*.example.info'
    http:
      paths:
      - backend:
          serviceName: ui-server
          servicePort: "80"
        path: /ci
  - host: '*.example.info'
    http:
      paths:
      - backend:
          serviceName: grafana
          servicePort: "3000"
        path: /grafana
  - host: '*.example.info'
    http:
      paths:
      - backend:
          headerRule:
          - Cache-Control public
          serviceName: ui-server
          servicePort: "80"
        path: /_org/assets
  - host: jenkins.example.com
    http:
      paths:
      - backend:
          serviceName: jenkins-service
          servicePort: "80"
  - host: api.example.com
    tcp:
    - backend:
        serviceName: apiserver
        servicePort: "50077"
      port: "3443"
      secretName: cert-example-com
  - host: api.example.com
    tcp:
    - backend:
        serviceName: example-apis
        servicePort: "50066"
      port: "3553"
      secretName: cert-example-com
  - host: diffusion.example.com
    tcp:
    - backend:
        serviceName: phabricator
        servicePort: "22"
      port: "22"
  tls:
  - hosts:
    - auth.example.com
    - maven.example.com
    - composer.example.com
    - npm.example.com
    - docker.example.com
    - api.example.com
    - jenkins.example.com
    - '*.example.info'
    secretName: cert-example-com
  - hosts:
    - '*.example.info'
    secretName: cert-example-info


Can i configure one frontend that uses one static port to route trafic to all the backends above or can i sett one singular nodeport to route traffic into a cluster with the above config.

**Comment by tamalsaha** [comment]
@colltrix35 , which version of Voyager are you using? can you show us the service that is created by voyager?

**Comment by colltrix35** [comment]
The latest version

**Comment by colltrix35** [comment]
I don't have a service yet i am still working  through the examples on github  and   got stuck here.

**Comment by tamalsaha** [comment]
Ok. See my reply earlier in this ticket: https://github.com/appscode/voyager/issues/400#issuecomment-323494395 .

Voyager will open a separate NodePort for each HAProxy port, not for each HTTP host. You can server multiple HTTP host from a single port. Please also note that this feature is going to be available in the next release of 3.2.0. You can try with beta image for now `appscode/voyager:3.2.0-beta.3`

**Comment by izutheo** [comment]
@tamalsaha  My question was because  of the sol you provided for @izutheo 
You explanation above is exactly what i am looking for but the configuration that you provided to izutheo above  seems to be relegated per host 

  I am not  a go program so i cannot really follow the code like other individuals here. so if you can point me to a documentation that can help with that configuration, it would be helpful.
